EdaYeo iP

nuagetsu opened the pull request:
# Bunbun
> “Your mind is for having ideas, not holding them.” – David Allen

Bunbun frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

nuagetsu committed:
feat: rename, greet, exit

nuagetsu committed:
feat: echo

nuagetsu committed:
feat: add, list

nuagetsu committed:
feat: Mark as Done

nuagetsu committed:
feat: to dos, events, deadlines

nuagetsu committed:
feat: automated text UI testing

nuagetsu committed:
feat: error handling

nuagetsu committed:
feat: delete

nuagetsu committed:
Add save to hard disk

nuagetsu committed:
Add date time handling

nuagetsu committed:
Add more OOP

nuagetsu committed:
Refactor with pacakge

nuagetsu committed:
Refactor runtest.bat

nuagetsu committed:
Merge branch 'add-gradle-support'

nuagetsu committed:
Add gradle support

nuagetsu committed:
Add command class

nuagetsu committed:
Add exceptions

nuagetsu committed:
Add Junit test cases

nuagetsu committed:
Change some output text

nuagetsu committed:
Add and reformat JavaDocs

nuagetsu committed:
Style change for coding standard

nuagetsu committed:
Add find command

nuagetsu committed:
Merge branch-A-CodingStandard

nuagetsu committed:
Merge branch-Level-9

nuagetsu committed:
Add a GUI for Bunbun

JovenSoh reviewed (commented):
Great code that follows good coding practices and is very well-documented via javaDocs

SSirMentos commented on src/main/java/Bunbun/Bunbun.java:
```diff
@@ -0,0 +1,61 @@
+    public Bunbun(String filePath) {
+        this.storage = new Storage(filePath);
+        this.ui = new Ui();
+        this.list = new TaskList(this.storage, this.ui);
```
LGTM. I like how you’ve organised the Bunbun class with clear separation of responsibilities for Storage, TaskList, and Ui. It keeps the code modular and easier to maintain.

SSirMentos commented on src/main/java/Bunbun/Bunbun.java:
```diff
@@ -0,0 +1,61 @@
+            String msg = Parser.getMessage();
+            if (msg.equals("bye")) {
+                this.storage.writeAllFromList(list);
+                this.ui.endScreen();
```
Good use of function names, makes the code readable and easy to understand!

JovenSoh commented on src/main/java/Bunbun/Gui/MainWindow.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    /** Injects the Bunbun instance */
+    public void setBunbun (Bunbun b) {
```
Maybe we could remove the space before the opening parenthesis

SSirMentos commented on src/main/java/Bunbun/Bunbun.java:
```diff
@@ -0,0 +1,61 @@
+ * @author Eda Yeo
+ * @version CS2103T AY24/25 Semester 1
+ */
+public class Bunbun {
```
Maybe include some javadocs for each function

SSirMentos commented on src/main/java/Bunbun/Bunbun.java:
```diff
@@ -0,0 +1,61 @@
+
+    public void run() {
+        this.ui.startScreen();
+        storage.initializeTaskFile();
```
The variables this.storage, this.ui, and this.list are consistently using the this. keyword for instance variables, which is a good practice. To maintain consistency, ensure that all instance variables throughout the code follow this convention.

SSirMentos reviewed (commented):
LGTM. Just some nits to fix

nuagetsu committed:
Add assertions

Assertions are missing.

Assertions play an important role in guarenteeing the
logic of code is sound.

Let's add assertions at appropriate parts.

Adding assertions are a good way to catch errors
in logic and fix them as quickly as possible.

nuagetsu committed:
Refactor to improve code quality

SLAP is weak in Command class and multiple functions
in TaskList class have deeply nested code.

Weak SLAP and deeply nested code leads to lower clarity.

Let's:
* Apply SLAP to the Command class by passing error
handling to the TaskList methods
* Refactor methods in TaskList class with guard
clauses and remove deep nesting

Having SLAP and removing deep nesting is desirable
as it improves clarity.

nuagetsu committed:
Merge pull request #1 from EdaYeo/branch-A-Assertions

Add assertions

nuagetsu committed:
Merge branch 'master' into branch-A-CodeQuality

nuagetsu committed:
Merge pull request #2 from EdaYeo/branch-A-CodeQuality

Refactor to improve code quality

nuagetsu committed:
Debug file initialisation

nuagetsu committed:
Add time-boxed tasks

nuagetsu committed:
Add CI

nuagetsu committed:
Add CI

nuagetsu committed:
Refactor with renaming

nuagetsu committed:
Add user guide

nuagetsu committed:
Update README.md

nuagetsu committed:
Update README.md

nuagetsu committed:
Update README.md

nuagetsu committed:
Debug build file

nuagetsu committed:
Merge branch 'master' of https://github.com/EdaYeo/ip

nuagetsu committed:
Debug filepath

nuagetsu committed:
Debug file initialization
